pkg/profiling/task/network/analyze/layer7/protocols/base/tracing.go

```diff
@@ -15,23 +15,32 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package protocols
+package base
 
 import (
 	"encoding/base64"
 	"fmt"
 	"strings"
+
+	v3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
 )
 
 type TracingContext interface {
 	TraceID() string
-	Provider() string
+	TraceSegmentID() string
+	SpanID() string
+	Provider() *TraceContextProvider
+}
+
+type TraceContextProvider struct {
+	Type v3.SpanAttachedEvent_SpanReferenceType
+	Name string
 }
 
 type SkyWalkingTracingContext struct {
 	TraceID0              string
-	SegmentID             string
-	SpanID                string
+	SegmentID0            string
+	SpanID0               string
```

Comment:
Why ends of `0`?

Reply:
That's used to avoid conflict between field and method.